src/gdb_main.c

```diff
@@ -423,35 +423,37 @@ static void exec_q_supported(const char *packet, const size_t length)
 
 static void exec_q_memory_map(const char *packet, const size_t length)
 {
+	target_s *target = cur_target;
```

Comment:
Seeing as packet is used in this function, please drop that `(void)packet;` line. Additionally, we have been trying to put all parameter use suppression before the body of the function so please put this new line after the line that turns off the warning that `length`is unused

Reply:
I'm trying to stick with the style of putting declarations at the top of the function, but I see now that the rest of the code mixes declarations with statements, so I will try to match that. (The `(void)packet;` was pre-existing, but I'll remove it.)

Reply:
C89 style decls are bugprone + pessimising and this is a C11 code base, so we've been trying to migrate it away from that style to prevent and fix bugs :) (there have been several caused by read-before-init and such things)